### DISCUSSION TITLE
No Software-Serial

### DISCUSSION THREAD

**Discussion by Battlecake91** [arendst/Tasmota#22431] [Support request] [answered]
Hello everyone,

I have problems with software-serial on a ESP32-C3 and ESP8266. No matter what I do, I don't manage to get the sserial-commands. 

As I understand from several discussions, I have two configure serial tx AND serial rx to get the function working. 
I configure the GPIOs like this:
![image](https://github.com/user-attachments/assets/1ad3826f-73ce-443f-b02e-0c2246375f2c)

But if I try to configure sserial with "sserialconfig":

`16:29:42.502 CMD: sserialconfig`
`16:29:42.508 RSL: RESULT = {"Command":"Unknown","Input":"SSERIALCONFIG"}`

So I read in this discussion: https://github.com/arendst/Tasmota/issues/683 that maybe programming it with wrong mode (DIO) can cause this, so I reprogrammed the ESP8266 and the issue is the same. So I tried a fresh ESP32-C3, because I thought the ESP8266 can't handle it, but it is still the same issue. I used esptool with DOUT.

I flashed a precompiled Tasmota 14.3.0.

**Comment by sfromis** [comment, marked as answer]
For serial bridge, you should configure the pin pair as `SerBr Tx/Rx` instead of `Serial Tx/Rx`, this will be software serial if not selecting pins able to do hardware serial. On ESP32, all pins can do hardware serial, and software serial is not used for serial bridge.

Note that on ESP8266, it is preferable (if available) to use pins compatible with hardware serial with the serial bridge.

**Comment by Battlecake91** [comment]
Ah! Very nice! Thank you very much. 

I tried it with SerBr TX/RX and it works.

I prefer Hardware-Serial, but I have some special problems. I try to communicate via D-Bus with a dishwasher from Siemens, it needs an open-collector/open-drain on the bus. And unfortunately the ESPs won't boot with something on the TX pin pulling down the signal before boot. So I have to do something in hardware or I just use software-Serial.

But the hint that the ESP32 can switch his hw-serial to every pin is very helpful. But this seems not be a possible workaround for the ESP8266. Anyway, software-serial is more than I need.

Thank you very much!!

**Reply by sfromis** [reply]
On the ESP8266, you also can have hardware serial on gpio 15/13, instead of the standard gpio 1/3.

BTW. Using gpio 0/2 for serial still runs into the risk of these strapping pins influencing boot.

**Reply by Battlecake91** [reply]
Thanks for the tip. Unfortunately I'm using a ESP-02S for my project and there is no gpio 15 on the outside.
I'm not 100% sure if I will use the ESP-02S for the final solution or making my own pcba just to plug it onto the bus-port. If I do this, I guess I will use a C3 or so anyway. But first I wanted to test if I can read stuff from my dishwasher with tasmota and make it useable.
